File: samples/source/fortran/hello.f90
! Simple hello world program in Fortran with some complexity for analysis
program hello_world
    implicit none

    ! Variable declarations
    integer :: i, sum_len
    integer :: num_args
    character(len=100) :: arg
    integer, parameter :: global_counter = 42

    ! Print hello message
    print *, "Hello, World from Fortran!"

    ! Get command line arguments
    num_args = command_argument_count()

    ! Calculate sum of argument lengths
    sum_len = 0
    do i = 1, num_args
        call get_command_argument(i, arg)
        sum_len = sum_len + len_trim(arg)
    end do

    ! Print results
    print *, "Number of arguments:", num_args
    print *, "Total argument length:", sum_len
    print *, "Global counter:", global_counter

    ! Call a subroutine
    call print_message("Fortran subroutine called")

contains

    ! Subroutine for additional complexity
    subroutine print_message(msg)
        character(len=*), intent(in) :: msg
        integer, save :: call_count = 0

        call_count = call_count + 1
        print *, msg
        print *, "Subroutine called", call_count, "times"
    end subroutine print_message

end program hello_world

File: samples/source/cpp/hello.cpp
#include <iostream>
#include <string>
#include <vector>
#include <memory>

// Simple hello world with some complexity for analysis
class HelloWorld {
private:
    std::string message;
    int counter;

public:
    HelloWorld(const std::string& msg = "Hello, World from C++!") : message(msg), counter(0) {}

    void printMessage() {
        std::cout << message << std::endl;
        counter++;
    }

    int getCounter() const { return counter; }
};

int main(int argc, char* argv[]) {
    HelloWorld hw;
    hw.printMessage();

    // Some basic operations to create interesting disassembly
    int sum = 0;
    std::vector<std::string> args(argv, argv + argc);

    for (const auto& arg : args) {
        sum += arg.length();
    }

    // Use smart pointer
    auto printer = std::make_unique<HelloWorld>("Sum printer");
    printer->printMessage();

    std::cout << "Total argument length: " << sum << std::endl;
    std::cout << "Counter value: " << hw.getCounter() << std::endl;

    return 0;
}

// Global variable
int global_counter = 42;

// Static function
static void static_function() {
    static int static_var = 0;
    static_var++;
    std::cout << "Static function called " << static_var << " times" << std::endl;
}